ocaml/testsuite/tests/typing-unique/overwriting_proj_push_down_bug.ml

```diff
@@ -328,7 +328,55 @@ let swap_inner (t : tree) =
 val swap_inner : tree -> tree @@ global many = <fun>
 |}]
 
-(* TODO: Update this test once overwriting is fully implemented.
+let match_guard r =
+  match r with
+  | { y } when String.equal y "" ->
+    let r = aliased_use r in
+    (r, y)
+  | { y } ->
+    let r = unique_use r in
+    (r, y)
+[%%expect{|
+(let
+  (unique_use/283 = (apply (field_imm 0 (global Toploop!)) "unique_use")
+   aliased_use/280 = (apply (field_imm 0 (global Toploop!)) "aliased_use")
+   match_guard/376 =
+     (function {nlocal = 0} r/378[(consts ()) (non_consts ([0: *, *]))]
+       [(consts ())
+        (non_consts ([0: [(consts ()) (non_consts ([0: *, *]))], *]))]
+       (let (y/379 =o (field_mut 1 r/378))
+         (if (apply (field_imm 8 (global Stdlib__String!)) y/379 "")
+           (let
+             (r/450 =[(consts ()) (non_consts ([0: *, *]))]
+                (apply aliased_use/280 r/378))
+             (makeblock 0 ([(consts ()) (non_consts ([0: *, *]))],*) r/450
+               y/379))
+           (let
+             (y/380 =o (field_mut 1 r/378)
+              r/451 =[(consts ()) (non_consts ([0: *, *]))]
+                (apply unique_use/283 r/378))
+             (makeblock 0 ([(consts ()) (non_consts ([0: *, *]))],*) r/451
+               y/380))))))
+  (apply (field_imm 1 (global Toploop!)) "match_guard" match_guard/376))
+val match_guard : unique_ record -> record * string @@ global many = <fun>
+|}]
+
+let match_guard_unique (unique_ r) =
+  match r with
+  | { y } when String.equal ((unique_use r).x) "" -> y
+  | _ -> ""
+[%%expect{|
+Line 3, characters 4-9:
+3 |   | { y } when String.equal ((unique_use r).x) "" -> y
+        ^^^^^
+Error: This value is read from here, but it has already been used as unique:
```

Comment:
Technically, both usages happen at the same time since the evaluation order is not determined. I have adjusted the error message slightly to reflect this; feel free to suggest a different error if that would be clearer.